### PR TITLE
Plans 2023: Domain upsell plans page redone

### PR DESCRIPTION
**Pull request by chriskmnds**
Fixes Automattic/martech#1551
Related to https://github.com/Automattic/martech/issues/1491
Based off https://github.com/Automattic/wp-calypso/pull/73421 & https://github.com/Automattic/wp-calypso/pull/73418
## Proposed Changes

Various updates for the domain upsell (`/plans` page with a `?get_domain` parameter) redesign. 
- Issue https://github.com/Automattic/martech/issues/1491
- Figma xkhhUlDowF13dRoXJqlRDM-fi-1:95 

cc @lucasmendes-design 

### old plans grid

Screenshot 2023-02-16 at 1 17 16 PM

### new plans grid

Screenshot 2023-02-16 at 1 18 17 PM

### clicking on "skip" or "keep my plan"

Screenshot 2023-02-16 at 1 31 01 PM

## Testing Instructions

* Go to `plans/[SITE]?get_domain=foobar` with and without the feature flag (`&flags=-onboarding/2023-pricing-grid`). This is equivalent to clicking on the Home page pannel.
* Confirm the behavior outlined in the issue (screenshots above).
* Confirm that everything else is sane without the `get_domain` parameter.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plans/components/plans-header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import { useTranslate } from 'i18n-calypso';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { WPCOM_PLANS_UI_STORE } from '../../store';
+import './style.scss';
+
+const DomainUpsellHeader: React.FunctionComponent = () => {
+	const { setShowDomainUpsellDialog } = useDispatch( WPCOM_PLANS_UI_STORE );
+	const translate = useTranslate();
+	const is2023OnboardingPricingGrid = isEnabled( 'onboarding/2023-pricing-grid' );
```
Comment (chriskmnds) [inline client/my-sites/plans/components/plans-header/index.tsx]: This needs changing to use the utility method.

**Comment by southp** [comment]
A quick note that I've gone through the flow both by appending the query arg directly and by accessing it from the upselling card with and without the feature flag.
image

And I don't spot any issue under desktop / mobile / tablet breakpoints. I'll first review the other data store PRs and then come back to this.

File: client/my-sites/plans/components/domain-upsell-dialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	const onContinuePlanClick = useCallback( () => {
+		onCloseDialog();
+		recordTracksEvent( 'calypso_plans_page_domain_upsell_continue_plan_click' );
+	}, [ onCloseDialog ] );
+
+	const buttons = [
+		{ action: 'cancel', label: translate( 'That works for me' ), onClick: onCancelPlanClick },
+		{
+			action: 'delete',
```
Comment (southp) [inline client/my-sites/plans/components/domain-upsell-dialog/index.tsx:42]: Non-blocking: 
I know it's been this way, but the action slug doesn't align with accepting the upsell offer. Could we remove this? From the `<Dialog>` code: https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/dialog/index.tsx#L47, the action slug won't do anything without a `onClose()` event handler anyway.

**Review by southp — APPROVED** [review]
The code in general looks good. 🚢 

I don't really know what's the right flow to make `domainSidebarExperimentUser` to be `true`.  so I hardcoded the value and saw the broken layout:

image

The old plan grid shows the same:
image

It's also broken in trunk now. Given that whatever fix we come up will be based on this, let's ship this first and fix the layout as a follow-up.

File: client/my-sites/plans/components/domain-upsell-dialog/index.tsx (hunk 2 above)
Comment (chriskmnds) [reply client/my-sites/plans/components/domain-upsell-dialog/index.tsx:42]: TS will complain if we don't / it's a nonoptional prop on BaseButton apparently. Happy to change to something else if you prefer. something other than "cancel" maybe as it's used for the other?

**Comment by chriskmnds** [comment]
@southp  thanks for taking the time. I don't have anything to add here, other than ship once https://github.com/Automattic/wp-calypso/pull/73421 gets approval. The PR needs another sanity check after renaming the folder in the base PR.

> It's also broken in trunk now. Given that whatever fix we come up will be based on this, let's ship this first and fix the layout as a follow-up.

Seeing the same on my end. Maybe one for @Automattic/explorers to [take a quick look](https://github.com/Automattic/wp-calypso/pull/73230#pullrequestreview-1302721763)? It might be the way we activate the `domainSidebarExperimentUser` flag.

**Review by paulopmt1 — APPROVED** [review]
Thanks for pinging us @chriskmnds! I did a quick test here and everything seems fine. 
We use this strategy to enable experiments simulating real user flow: pb5gDS-2nD-p2#manually-assigning-a-user-to-an-experiment

We're using this experiment: `calypso_sidebar_upsell_dedicated_upgrade_flow_202301`

Both `treatment` and `control` are working great for me:

Treatment: 

image


Control:

image

Please let me know if we can assist with further tests or implementation :)

**Comment by paulopmt1** [comment]
The broken tests with `domainSidebarExperimentUser` is caused because the layout [also needs to understand](https://github.com/Automattic/wp-calypso/blob/update/plans-grid-2023-header-nav-domain-upsell/client/layout/index.jsx#L385) if we're at the experiment or not, so for changing the experiment variation by code you can change [the isDomainSidebarExperimentUser selector](https://github.com/Automattic/wp-calypso/blob/update/plans-grid-2023-header-nav-domain-upsell/client/state/selectors/is-domain-sidebar-experiment-user.js#L10).

**Comment by paulopmt1** [comment]
It seems we can't select the Free plan at the domain upsell flow @chriskmnds. Can you please check?

image

**Comment by chriskmnds** [comment]
@paulopmt1 Thanks for testing. 

> It seems we can't select the Free plan at the domain upsell flow @chriskmnds. Can you please check?

This also happens in trunk for me right now (nothing happens when clicking on "start with free" - the rest of the buttons work fine). I will rebase and try again, maybe figure this out - but it looks like something outside of this PR 🤔

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7830324

Thank you @chriskmnds for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7830324).

**Pull request merged**